pkg/cli/host_preflight.go

```diff
@@ -26,32 +28,49 @@ func NewHostPreflightCmd(cli CLI) *cobra.Command {
 		SilenceUsage: true,
 		Args:         cobra.ExactArgs(1),
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			return viper.BindPFlags(cmd.PersistentFlags())
+			return cli.GetViper().BindPFlags(cmd.PersistentFlags())
 		},
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			return viper.BindPFlags(cmd.Flags())
+			return cli.GetViper().BindPFlags(cmd.Flags())
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
+			v := cli.GetViper()
+
 			installerSpec, err := installer.RetrieveSpec(cli.GetFS(), args[0])
 			if err != nil {
 				return errors.Wrap(err, "retrieve installer spec")
 			}
 
-			builtin := preflight.Builtin()
 			data := installer.TemplateData{
 				Installer: *installerSpec,
-				IsPrimary: viper.GetBool("is-primary"),
-				IsJoin:    viper.GetBool("is-join"),
-				IsUpgrade: viper.GetBool("is-upgrade"),
+				IsPrimary: v.GetBool("is-primary"),
+				IsJoin:    v.GetBool("is-join"),
+				IsUpgrade: v.GetBool("is-upgrade"),
 			}
-			spec, err := installer.ExecuteTemplate("installerSpec", builtin, data)
+
+			builtin := preflight.Builtin()
+			preflightSpec, err := decodePreflightSpec(builtin, data)
 			if err != nil {
-				return errors.Wrap(err, "execute installer template")
+				return errors.Wrap(err, "builtin")
 			}
 
-			preflightSpec, err := preflight.Decode(spec)
-			if err != nil {
-				return errors.Wrap(err, "decode spec")
+			for _, filename := range v.GetStringSlice("spec") {
```

Comment:
Could you comment on what's going on in the loop? I don't understand why the maybe append functions are comparing titles.

Reply:
I was thinking about adding registry http checks and thought there would be a lot of duplication since say lots of addons use docker hub. I am trying to avoid this but there are no ids in collectors and analyzers. I could make a change to troubleshoot to check that the checks are equivalent. What do you prefer here? I will additionally add a comment once we have an agreed upon solution.

Reply:
I see.  I think a comment here will be fine.

Reply:
The more i think about the approach here the more i dislike it. I may try to think of another solution.